wandb/cli/cli.py

```diff
@@ -1000,34 +996,43 @@ def launch_sweep(
             wandb.termerror(f"Failed to load job. Error: {e}")
             return False
 
-    entrypoint, args = sweep_utils.construct_scheduler_entrypoint(
+    args = sweep_utils.construct_scheduler_args(
         sweep_config=parsed_sweep_config,
         queue=queue,
         project=project,
-        num_workers=scheduler_args.get("num_workers", 8),
         author=entity,
     )
-    if not entrypoint:
+    if not args:
         # error already logged
         return
 
+    overrides = {
+        "overrides": {
+            "run_config": {
+                "scheduler": scheduler_args,
+                "launch": launch_args,
+            },
+            "args": args,
+        }
+    }
+
     # Launch job spec for the Scheduler
     launch_scheduler_spec = construct_launch_spec(
-        uri=SCHEDULER_URI,
+        uri=Scheduler.PLACEHOLDER_URI,
         api=api,
         name="Scheduler.WANDB_SWEEP_ID",
         project=project,
         entity=entity,
         docker_image=scheduler_args.get("docker_image"),
         resource=scheduler_args.get("resource", "local-process"),
-        entry_point=entrypoint,
+        entry_point=Scheduler.ENTRYPOINT,
         resource_args=scheduler_args.get("resource_args", {}),
         repository=launch_args.get("registry", {}).get("url", None),
         job=None,
         version=None,
-        launch_config={"overrides": {"args": args}},
-        run_id=None,
-        author=None,  # author gets passed into scheduler command
+        launch_config=overrides,
+        run_id="WANDB_SWEEP_ID",
```

Comment:
Wait, this is confusing me now. Why do we make this WANDB_SWEEP_ID? Couldn't the scheduler upsert sweep, get the sweep id from that and use that to init the run id?

Reply:
The `WANDB_SWEEP_ID` just gets macrod in the backend to the sweep ID during upsertSweep. Same as the `Scheduler. WANDB_SWEEP_ID` above. Then we use that in the run_config of the job to init the wandb run in the scheduler. Maybe I don't understand.... @KyleGoyette